polkadot/runtime/parachains/src/disputes/slashing.rs

```diff
@@ -547,14 +557,26 @@ impl<T: Config> Pallet<T> {
 		let _ = <UnappliedSlashes<T>>::clear_prefix(old_session, REMOVE_LIMIT, None);
 	}
 
-	pub(crate) fn unapplied_slashes() -> Vec<(SessionIndex, CandidateHash, PendingSlashes)> {
-		<UnappliedSlashes<T>>::iter().collect()
+	pub(crate) fn unapplied_slashes() -> Vec<(SessionIndex, CandidateHash, PendingSlashesV1)> {
+		// Converting UnappliedSlashes to use the old SlashingOffence enum
+		// instead of the new DisputeOffenceKind enum to maintain the same
+		// behavior for the runtime api.
+		<UnappliedSlashes<T>>::iter()
+			.filter_map(|(session, candidate_hash, slash_v2)| {
+				match PendingSlashesV1::try_from(slash_v2) {
+					Ok(slash_v1) => Some((session, candidate_hash, slash_v1)),
+					Err(_) => None, // Skip if conversion fails (e.g., variant not representable in old format)
```

Comment:
this should be logged probably, although during the migration period while we spam spamming approval checkers, but the runtime api is not upgraded yet, this could be spammy - maybe an info level?